tests/trainers/test_instance_segmentation.py

```diff
@@ -0,0 +1,165 @@
+
+
+# TODO: This is not even used yet
+class InstanceSegmentationTestModel(Module):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__()
+        self.fc = nn.Linear(1, 1)
+
+    def forward(self, images: Any, targets: Any = None) -> Any:
+        batch_size = len(images)
+        if self.training:
+            assert batch_size == len(targets)
+            # use the Linear layer to generate a tensor that has a gradient
+            return {
+                'loss_classifier': self.fc(torch.rand(1)),
+                'loss_box_reg': self.fc(torch.rand(1)),
+                'loss_objectness': self.fc(torch.rand(1)),
+                'loss_rpn_box_reg': self.fc(torch.rand(1)),
+            }
+        else:  # eval mode
+            output = []
+            for i in range(batch_size):
+                boxes = torch.rand(10, 4)
+                # xmax, ymax must be larger than xmin, ymin
+                boxes[:, 2:] += 1
+                output.append(
+                    {
+                        'boxes': boxes,
+                        'masks': torch.randint(2, images.shape[1:]),
+                        'labels': torch.randint(2, (10,)),
+                        'scores': torch.rand(10),
+                    }
+                )
+            return output
+
+
+def plot(*args: Any, **kwargs: Any) -> None:
+    return None
+
+
+def plot_missing_bands(*args: Any, **kwargs: Any) -> None:
+    raise RGBBandsMissingError()
+
+
+class TestInstanceSegmentationTask:
+    @pytest.mark.parametrize('name', ['ftw_ins_seg', 'vhr10_ins_seg'])
+    def test_trainer(
+        self, monkeypatch: MonkeyPatch, name: str, fast_dev_run: bool
+    ) -> None:
+        match name:
+            case 'ftw_ins_seg':
+                pytest.importorskip('pyarrow')
+
+        config = os.path.join('tests', 'conf', name + '.yaml')
+
+        args = [
+            '--config',
+            config,
+            '--trainer.accelerator',
+            'cpu',
+            '--trainer.fast_dev_run',
+            str(fast_dev_run),
+            '--trainer.max_epochs',
+            '1',
+            '--trainer.log_every_n_steps',
+            '1',
+        ]
+
+        main(['fit', *args])
+        try:
+            main(['test', *args])
+        except MisconfigurationException:
+            pass
+        try:
+            main(['predict', *args])
+        except MisconfigurationException:
+            pass
+
+    def test_invalid_model(self) -> None:
+        match = 'Invalid model type'
+        with pytest.raises(ValueError, match=match):
+            InstanceSegmentationTask(model='invalid_model')
+
+    def test_invalid_backbone(self) -> None:
+        match = 'Invalid backbone type'
+        with pytest.raises(ValueError, match=match):
+            InstanceSegmentationTask(backbone='invalid_backbone')
+
+    def test_weights(self) -> None:
+        InstanceSegmentationTask(weights=True, num_classes=91)
```

Comment:
Yet another test that downloads data on the fly (#1088), but let's fix that another day when we figure out how to support custom weights.